7_react/react1/week2/my-app/src/App.js

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useEffect } from "react";
+import "./App.css";
+import { v4 as uuidv4 } from "uuid";
+
+const todos = [
+  {
+    id: 1,
+    description: "Get out of bed",
+  },
+  {
+    id: 2,
+    description: "Brush teeth",
+  },
+  {
+    id: 3,
+    description: "Eat breakfast",
+  },
+];
+
+export default function App() {
+  const [list, setList] = useState(todos);
+  const [counter, setCounter] = useState(0);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setCounter((prevCounter) => prevCounter + 1);
+    }, 1000);
+
+    return () => clearInterval(interval);
+  }, []);
+
+  return (
+    <div>
+      <h1>Todolist</h1>
+      <p>You have used {counter} seconds on this website</p>
+      <button
+        onClick={() =>
+          setList([...list, { id: uuidv4(), description: "random text" }])
+        }
+      >
+        Add todo
+      </button>
+      <ul>
+      {list.length > 0 ? (
+        list.map((listItem) => {
+          return (
+            <li key={listItem.id} className="list-item">
+              <input type="checkbox" id={listItem.id} />
+              <label for={listItem.id} className="strikethrough">
+                {listItem.description}
+              </label>
+              <button
+                onClick={() => {
+                  setList(list.filter((e) => e.id !== listItem.id));
```

Comment:
Nice and simple way of deleting an item. Good approach.

At a glance, it may not be obvious at a to all developers what `setList(list.filter((e) => e.id !== listItem.id))` does. You could consider to add a method to make the code easier to read for other developers and your future self:

``` js
const removeTodo = (todoId) {
    setList(list.filter((e) => e.id !== todoId))
}
```
then you can call it like this:
``` js
<button onClick={() => removeTodo(listItem.id)}/>
```

This is not a fixed rule but more of a personal preference you (and your team) would have to decide. 
* One developer could argue that the extra method would just bloat the code and be unnecessary 😞 
* Another developer always finds your code the easiest to understand and hail you as the best developer in the team. 🥇